src/main/java/com/amazon/ion/impl/IonRawTextWriter_1_1.kt

```diff
@@ -0,0 +1,365 @@
+package com.amazon.ion.impl
+
+import com.amazon.ion.*
+import com.amazon.ion.impl.IonRawTextWriter_1_1.ContainerType.*
+import com.amazon.ion.util.*
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.time.Instant
+
+/**
+ * A raw writer for Ion 1.1 text. This should be combined with managed writer to handle concerns such as macros and
+ * possible symbol interning.
+ *
+ * Notes:
+ *  - Never writes using "long string" syntax in order to simplify the writer.
+ *  - Uses `[: ... ]` for expression groups.
+ *  - Does not try to resolve symbol tokens. That is the concern of the managed writer.
+ */
+class IonRawTextWriter_1_1 internal constructor(
+    private val options: _Private_IonTextWriterBuilder,
+    private val output: _Private_IonTextAppender,
+) : IonRawWriter_1_1 {
+
+    enum class ContainerType {
+        List,
+        SExp,
+        Struct,
+        Macro,
+        ExpressionGroup,
+        Top,
+    }
+
+    private var closed = false
+
+    private val ancestorContainersStack: ArrayList<ContainerType> = ArrayList()
+    private var currentContainer: ContainerType = Top
+    private var currentContainerHasValues = false
+
+    private var isPendingSeparator = false
+    private var isPendingLeadingWhitespace = false
+
+    private var fieldNameTextBuffer: CharSequence? = null
+    private var fieldNameIdBuffer: Int = -1
```

Comment:
What's the purpose of the "Buffer" suffix?

Reply:
It's meant to parallel the annotations buffer `val`s. (Technically, these could be considered a fixed-size buffer.) Maybe "buffer" is misleading. Do you have a preference for a different name?

Reply:
Are they not just `fieldNameText` and `fieldNameId`?